stringManipulation/src/stringManipulation.cpp

```diff
@@ -5,7 +5,114 @@ namespace LLGE
 	namespace stringManipulation
 	{
 		//when implementing this functions try to not repeat yourself
+		size_t stringManipulation::findChar(std::string_view source, char character)
+		{
 
+			size_t index = 0;
+			for (; index <= source.size(); index++)
+			{
+				if (source[index] == character)
+				{
+					break;
+				}
+			}
+			if (index > source.size())
+			{
+				return -1;
+			}
+			return index;
+		}
+		void stringManipulation::toLower(std::string_view source, std::string& dest)
+		{
+			size_t index=0;
+			while(index<=source.size())
+			{
+
+				if ((int)source[index] >= 65 && (int)source[index] <= 90)
+				{
+					dest+= (char)(((int)source[index]) + 32);
+
+				}
+				else {
+					dest += source[index];
+				}
+				index++;
+			}
+		}
+
+		void stringManipulation::toUpper(std::string_view source, std::string& dest)
+		{
+			size_t index = 0;
+
+			while (index <= source.size())
+			{
+				if ((int)source[index] >= 97 && (int)source[index] <= 122)
+				{
+					dest += (char)(((int)source[index]) - 32);
+				}
+				else {
+					dest += source[index];
+				}
+				index++;
+			}
+
+		}
+
+		void stringManipulation::toLower(std::string_view source, char* dest, size_t destSize)
+		{
+			size_t index = 0;
+			while (index < destSize && index <= source.size())
+			{
+				if ((int)source[index] >= 65 && (int)source[index] <= 90)
+					*(dest + index) = (char)(((int)source[index]) + 32);
+				else
+					*(dest + index) = source[index];
+
+				index++;
+			}
+		}
+		void stringManipulation::toUpper(std::string_view source, char* dest, size_t destSize) 
+		{
+			size_t index = 0;
+
+			while (index < destSize && index <= source.size())
+			{
+				if ((int)source[index] >= 97 && (int)source[index] <= 122)
+					*(dest+index)=(char)(((int)source[index]) - 32);
+				else
+					*(dest+index)= source[index];
+
+				index++;
+			}
+		}
+
+		void stringManipulation::removeCharacters(std::string_view source, char* dest, std::string_view charsToRemove, size_t destSize)
```

Comment:
hm this can definitely be implemented a lot more easily, check this out

https://github.com/meemknight/PikaEngine/blob/f52166b49cafd2cab99bf9f901f682aa0fe54e02/Pika/core/pikaSTD/stringManipulation/stringManipulation.cpp#L7